src/sagas/auth.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import { put, call, all, takeLatest } from "redux-saga/effects";
+import type { Saga } from "redux-saga";
+import { clearProfile } from "../actions/profile";
+import {
+  authRequest,
+  authSuccess,
+  authError,
+  authDisable,
+  type Login
+} from "../actions/auth";
+import {
+  LOGIN,
+  LOGOUT,
+  UNAUTHORIZED_ERROR_401
```

Comment:
I think you don't need it. Just add global interceptor to axios or fetch and handle error globally.
If error with status 401 returned - just dispatch 'logout' action and redirect the user to home page.